vendor/github.com/docker/distribution/registry/client/blob_writer.go

```diff
@@ -99,7 +99,7 @@ func (hbu *httpBlobUpload) Write(p []byte) (n int, err error) {
 		return 0, fmt.Errorf("bad range format: %s", rng)
 	}
 
-	return (end - start + 1), nil
```

Comment:
Do you mean MultiWriter? I can't find Multi{Reader,Writer} in this PR, but in general `io.Copy(io.MutliWriter(blobWriter, digester), tarStream)` can be replaced by `blobWriter.ReadFrom(io.TeeReader(tarStream, digester))`.